lua/autorun/server/sv_restrict_propcore.lua

```diff
@@ -22,24 +22,46 @@ function restrictPropCoreFunctions()
         --"propBreak(e:)"
     }
 
-    for _, signature in pairs( restrictedFunctions ) do
-        if wire_expression2_funcs then
-            local oldFunc = wire_expression2_funcs[signature][3]
-
-            wire_expression2_funcs[signature][3] = function( self, ... )
-                if disallowedRanks[self.player:GetUserGroup()] == nil then
-                    local isInBuildMode = self.player:GetNWBool("CFC_PvP_Mode", false) == false
-
-                    if isInBuildMode or self.player:IsAdmin() then
-                        return oldFunc( self, ... )
-                    else
-                        self.player:ChatPrint( "You can't use PropCore in PvP mode" )
-                    end
-                else
-                    self.player:ChatPrint( "You don't have access to " .. signature )
-                end
+
+    local adminOnlyFunctions = {
+        "use(e:)"
+    }
+
+    restrict( restrictedFunctions, function( self,  ... )
```

Comment:
Maybe we could extract the anonymous functions in this call out to their own named functions.